pkg/query/querier.go

```diff
@@ -345,8 +354,11 @@ func (q *querier) selectFn(ctx context.Context, hints *storage.SelectHints, ms .
 		}, nil
 	}
 
-	// TODO(fabxc): this could potentially pushed further down into the store API to make true streaming possible.
-	sortDedupLabels(resp.seriesSet, q.replicaLabels)
+	// Lazy retrieval strategy does not guarantee that series
+	// will be sorted with dedup labels at the end.
+	if q.proxy.RetrievalStrategy() == store.LazyRetrieval {
```

Comment:
And this is only necessary if some store sends us unsorted data, right? :thinking:

Reply:
Yes, I think this should only be necessary if any one store has unsorted data. I will do some tests with the new receivers soon.